.github/workflows/my-ci-pipeline.yml

```diff
@@ -16,3 +16,39 @@ jobs:
         - run: docker build --target test --tag todo_app:test .
         - run: docker run --env-file=.env.template todo_app:test todo_app/tests/unit
         - run: docker run --env-file=.env.template todo_app:test todo_app/tests/integration
+    push_to_registry:
+      needs: build
+      name: Push Docker image to Docker Hub
+      runs-on: ubuntu-latest
+      steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: rogvan/todoapp:prod
```

Comment:
Is there a reason for this action? I can see it can work nicely with the build-push action below, but I believe that relies on being used explicitly in that step, as in [the example in their docs](https://github.com/docker/metadata-action):

```
      -
        name: Build and push
        uses: docker/build-push-action@v4
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
```

Given that you're planning to tag it consistently regardless, it feels like the metadata isn't of huge value. Admittedly, I've not really used labels before so possibly I'm missing something there, [this article does make a reasonable case for them](https://medium.com/@chamilad/lets-make-your-docker-image-better-than-90-of-existing-ones-8b1e5de950d) but then I believe it's assuming you're setting some explicitly at some point!